lib/rock/webapp/tasks/api.rb

```diff
@@ -269,6 +275,51 @@ def get_port(name_service, name, port_name, init = false, timeout = 30)
                     end
                     
                     
+                    desc 'list operations'
+                    get ':name_service/:name/operations/' do
+                        taskhash = Hash[task_by_name(params[:name_service], params[:name]).to_h]
+                        model = taskhash[:model]
+                        Hash[operations: model[:operations]]
+                    end
+                    
+                    desc 'get operation parameters sample'
+                    get ':name_service/:name/operations/:operation/sample' do
+                        op = get_operation(params[:name_service], params[:name],params[:operation])
+                        paramarray = Array.new 
+                        op.arguments_types.each do |elem|
+                            entry = elem.new
+                            paramarray << entry.to_json_value(:special_float_values => :string)
+                        end
+                        Hash[:args => paramarray]
+                    end
+                    
+                    desc 'run operation'
+                    params do
+                        optional :blocking, type: String, default: "true"
+                    end
+                    post ':name_service/:name/operations/:operation' do
+                        op = get_operation(params[:name_service], params[:name],params[:operation])
+                            
+                        begin
+                            obj = MultiJson.load(request.params["value"])
+                        rescue MultiJson::ParseError => exception
+                            error! "malformed JSON string: #{request.params["value"]}", 415
+                        end
+                        
+                        begin
+                            params = obj["args"]
```

Comment:
It's working because JSON ends up being a mix of arrays and hashes, and Typelib accepts unmarshalling compounds/arrays/vectors based on such mix.

However, it's going to fail for e.g. base64-encoded values